orttraining/orttraining/python/training/ortmodule/_training_manager.py

```diff
@@ -246,7 +259,7 @@ def _create_execution_agent(self):
             C.OrtDevice(get_ort_device_type(self._device.type),
                         C.OrtDevice.default_memory(),
                         _utils.get_device_index(self._device)
-            )] * len(self._graph_info.user_output_names)
+            )] * (len(self._graph_info.user_output_names) + len(self._graph_info.frontier_node_arg_map))
```

Comment:
Do we need to add `len(self._graph_info.frontier_node_arg_map)` when grad accumulation is not enabled too?

Reply:
without gradient accumulation enabled, the map would be empty, hence this statement is safe.